lightningd/options.c

```diff
@@ -516,8 +516,29 @@ static char *opt_force_featureset(const char *optarg,
 				  struct lightningd *ld)
 {
 	char **parts = tal_strsplit(tmpctx, optarg, "/", STR_EMPTY_OK);
-	if (tal_count(parts) != NUM_FEATURE_PLACE)
-		return "Expected 5 feature sets (init, globalinit, node_announce, channel, bolt11) separated by /";
+	if (tal_count(parts) != NUM_FEATURE_PLACE) {
+		if (!strstarts(optarg, "-") && !strstarts(optarg, "+"))
+			return "Expected 5 feature sets (init, globalinit,"
+			       " node_announce, channel, bolt11) separated"
+			       " by / OR +/-<feature_num>";
+
+		char *endp;
+		long int n = strtol(optarg + 1, &endp, 10);
+		const struct feature_set *f;
+		if (*endp || endp == optarg + 1)
+			return "Invalid feature number";
+
+		f = feature_set_for_feature(NULL, n);
+		if (strstarts(optarg, "-")
+		    && !feature_set_sub(ld->our_features, take(f)))
+			return "Feature unknown";
+
+		if (strstarts(optarg, "+")
+		    && !feature_set_or(ld->our_features, take(f)))
+			return "Feature already flagged-on";
```

Comment:
This could lead to accidentally breaking configurations on upgrade. Should probably be a warning log message, and not a direct failure in the options parsing.

Reply:
Yes, the break is intentional. It's specifically aimed at breaking tests.  Since this is a dev- option, it's probably fine to notify anyone using it (via a start-failure) that their option defaults have changed.